crates/bevy_reflect/src/lib.rs

```diff
@@ -915,6 +896,38 @@ bevy_reflect::tests::should_reflect_debug::Test {
         assert_eq!(expected, format!("\n{:#?}", reflected));
     }
 
+    #[test]
+    fn reflect_type_name() {
+        #[derive(TypeName)]
+        struct Foo;
+        let name = Foo::name();
+        assert_eq!(name, "bevy_reflect::tests::Foo");
+
+        #[derive(TypeName)]
+        struct Goo<T: TypeName + 'static> {
```

Comment:
Is `'static` necessary?

Reply:
Yes, because `TypeName` use `GenericDataCell` that use `TypeId` that require `'static`

Reply:
Should it be a bound on `TypeName` itself then?

Reply:
Yes, make sense.